python/API_GW_CACHE_ENABLED_AND_ENCRYPTED/API_GW_CACHE_ENABLED_AND_ENCRYPTED.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Description:
+  Checks whether the API GW stage has cache enabled and is encrypted. The rule is NON_COMPLIANT if the API GW stage is not configured for cache and is not encrypted.
+
+Trigger:
+  Configuration Change
+
+Reports on:
+  AWS::ApiGateway::Stage
+
+Rule Parameters:
+  None
+
+Scenarios:
+  Scenario: 1
+     Given: The API GW stage caching is not enabled
+      Then: Return NON_COMPLIANT
+  Scenario: 2
+     Given: The API GW stage caching is enabled
+       And: The API GW stage caching is not encrypted
+      Then: Return NON_COMPLIANT
+  Scenario: 3
+     Given: The API GW stage caching is enabled
+       And: The API GW stage caching is encrypted
+      Then: Return COMPLIANT
+"""
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ApiGateway::Stage'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    stage = configuration_item['configuration']
+    # Scenario 1: If caching is not enabled for the stage, return NON_COMPLIANT
+    if not stage['cacheClusterEnabled']:
+        return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', 'Cache is not enabled.')
+    for method in stage['methodSettings']:
+        # Scenario 1: If caching is enabled for the stage, check if it is enabled for all methods
+        if not stage['methodSettings'][method]['cachingEnabled']:
+            return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', 'Cache is not enabled.')
+        # Scenario 2: If caching is enabled for the stage, check if it is encrypted for all methods
+        if not stage['methodSettings'][method]['cacheDataEncrypted']:
+            return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', 'Cache is not encrypted.')

```

Comment:
Please modify the annotation to 'The cache for this Amazon API Gateway Stage is not encrypted.'